src/solve/test.rs

```diff
@@ -536,21 +536,47 @@ fn elaborate_normalize() {
     test! {
         program {
             trait Eq { }
+            struct i32 { }
 
             trait Item where <Self as Item>::Out: Eq {
                 type Out;
             }
+
+            impl Eq for i32 { }
+            impl Item for i32 {
+                type Out = i32;
+            }
         }
 
         goal {
             forall<T, U> {
-                if (T: Item<Out = U>) {
+                if (WellFormed(T: Item), T: Item<Out = U>) {
```

Comment:
Here indeed the former suffices. At first I thought we didn't need something like WF(Projection). But now that you say it, there's indeed no way to have something like that:
```rust
trait Foo where <Self as Foo>::Item: Eq {
    type Item;
}

if (WellFormed(T: Foo<Item = Bar>)) { ... }
```
Currently, we have to add ```T: Foo<Item = Bar>``` into the environment in order to trigger the normalization, and rely on ```WF(T: Foo)```. But as a consequence we are also pushing ```T: Foo``` into the environnement (it is implied by ```T: Foo<Item = Bar>```) so basically we are making a stronger hypothesis that initially wanted.
It may be not so difficult to fix however.